SD-K9/src/app/providers/map-coordinator.service.ts

```diff
@@ -1,119 +1,240 @@
     }
+    // this._buildingEntry(parsedRoute.building)
+
+    private async _prepareIndoor(maps: MapItem[], parsedRoute: FloorPlanIdentifier, classID?: string, entryID?: string) {
+        maps.push(new MapItem(FloorPlanComponent, parsedRoute));
 
+        if (classID && entryID) {
+            // TODO: remove Location layer and simplify to reate SVGCoordinate directly
+            // Setup initial SVGCoordinate 
+            let initLocation: Location = new Location();
+            let iSvgCoordinate: SVGCoordinate = await this._svgManager.getClassroom(entryID, parsedRoute.building, parsedRoute.floor); // TODO: get building entry point from config
+            initLocation.setCoordinate(await iSvgCoordinate);
+            // Setup final SVGCoordinate
+            let finalLocation: Location = new Location();
+            let fSvgCoordinate: SVGCoordinate = await this._svgManager.getClassroom(classID, parsedRoute.building, parsedRoute.floor);
+            finalLocation.setCoordinate(await fSvgCoordinate);
+            // activate pathfinder
+            this._routeStore.storeRoute({id: parsedRoute.id, route: {source: initLocation, destination: finalLocation}});
+        }
+    }
+    
+    // TODO: delete when routes are extracted from map-coordinator to route-coordinator
+    async getRoute(iInitLocation: Location, iDestination: Location) {
+        this._routeCoordinator.getIndoorRoute(iInitLocation, iDestination);
+    }
+
+    // TODO: delete when routes are extracted from map-coordinator to route-coordinator
     setVerticalTransportationMode(mode: string) {
-        this._verticalTransportationMode = mode;
+        this._routeCoordinator.setVerticalTransportationMode(mode);
     }
 
+    // TODO: delete when routes are extracted from map-coordinator to route-coordinator
     async nextRoute() {
-        const routeNavigator: RouteNavigator = new RouteNavigator(this._indoorRouteBuilder);
-        routeNavigator.getRoute(this._routeLocationList[0].from, this._routeLocationList[0].to);
-        this._routeLocationList.shift();
-        if (this._routeLocationList.length === 0) {
-            this._hasNextRoute = false;
-        }
+        this._routeCoordinator.nextRoute();
     }
 
+    // TODO: delete when routes are extracted from map-coordinator to route-coordinator
     hasNextRoute() {
-        return this._hasNextRoute;
+        return this._routeCoordinator.hasNextRoute();
+    }
+
+    // TODO: replace with config
+    private _buildingPostalCode(building: string): string {
+        switch (building) {
+            case 'hall':
+                return "h3g1m8";
+            case 'loyola':
+                return "h4b1r6";
+        }
+    }
+
+    // TODO: replace with config
+    private _buildingEntry(building: string): string {
+        switch (building) {
+            case 'hall':
+                return "H-806";
+            case 'loyola':
+                return "L-101";
+        }
+    }
```

Comment:
Could these all go in a buildings.json alongside other info like coordinates for the outdoor map building overlays?

Reply:
Yes they can.
Or should we place them in firestore since we will need it for Google Calendar authentication. We can also store points of interests that are common for all users.
I also think the config/buildings.json/db should be done in another issue because it depends on many decisions.

Reply:
I opened #81 with the json config files in mind. Maybe we can start with that and then a migration to firestore once its setup will be easy since all the info will be at the same place?

Reply:
Sure, we can do that.